src/Discord.Net.Core/Entities/Interactions/Modals/ModalBuilder.cs

```diff
@@ -1,54 +1,53 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace Discord
 {
+    /// <summary>
+    ///     Represents a builder for creating a <see cref="Modal"/>.
+    /// </summary>
     public class ModalBuilder
     {
+        private string _customId;
+
         /// <summary>
-        ///     Gets or sets the components of the current modal.
+        ///     Creates a new instance of the <see cref="ModalBuilder"/>.
         /// </summary>
-        public ModalComponentBuilder Components { get; set; } = new();
+        /// <param name="title">The modal's title.</param>
+        /// <param name="customId">The modal's customId.</param>
+        /// <param name="components">The modal's components.</param>
+        /// <exception cref="ArgumentException">Only TextInputComponents are allowed.</exception>
+        public ModalBuilder(string title, string customId, ModalComponentBuilder components = null)
+        {
+            Title = title;
+            CustomId = customId;
+            Components = components ?? new();
+        }
 
         /// <summary>
         ///     Gets or sets the title of the current modal.
         /// </summary>
         public string Title { get; set; }
 
         /// <summary>
-        ///     Gets or sets the custom id of the current modal.
+        ///     Gets or sets the custom ID of the current modal.
         /// </summary>
         public string CustomId
         {
             get => _customId;
             set => _customId = value?.Length switch
             {
-                > ComponentBuilder.MaxCustomIdLength => throw new ArgumentOutOfRangeException(nameof(value), $"Custom Id length must be less or equal to {ComponentBuilder.MaxCustomIdLength}."),
-                0 => throw new ArgumentOutOfRangeException(nameof(value), "Custom Id length must be at least 1."),
+                ComponentBuilder.MaxCustomIdLength => throw new ArgumentOutOfRangeException(nameof(value), $"Custom ID length must be less or equal to {ComponentBuilder.MaxCustomIdLength}."),
+                0 => throw new ArgumentOutOfRangeException(nameof(value), "Custom ID length must be at least 1."),
                 _ => value
             };
         }
 
-        private string _customId;
-
-        public ModalBuilder() { }
-
```

Comment:
this is a breaking change

Reply:
https://github.com/discord-net/Discord.Net/pull/2773/commits/d7f4b1dadca8b2bcb96b8bae4ffb68138f88e716